librz/core/cmd/cmd_analysis.c

```diff
@@ -3095,143 +2994,122 @@ RZ_API void rz_cmd_analysis_calls(RzCore *core, const char *input, bool printCom
-					if (si) {
-						rz_cons_printf(".equ SYS_%s %s\n", si->name, syscallNumber(n));
-						rz_syscall_item_free(si);
-					} else
-						eprintf("Unknown syscall number\n");
-				} else {
-					n = rz_syscall_get_num(core->analysis->syscall, input + 3);
-					if (n != -1) {
-						rz_cons_printf(".equ SYS_%s %s\n", input + 3, syscallNumber(n));
-					} else {
-						eprintf("Unknown syscall name\n");
-					}
-				}
-			} else {
-				list = rz_syscall_list(core->analysis->syscall);
-				rz_list_foreach (list, iter, si) {
-					rz_cons_printf(".equ SYS_%s %s\n",
-						si->name, syscallNumber(si->num));
-				}
-				rz_list_free(list);
-			}
-		} else {
-			if (input[1] == ' ') {
-				if (!isalpha((ut8)input[2]) && (n = rz_num_math(num, input + 2)) >= 0) {
-					si = rz_syscall_get(core->analysis->syscall, n, -1);
-					if (si) {
-						rz_cons_printf("#define SYS_%s %s\n", si->name, syscallNumber(n));
-						rz_syscall_item_free(si);
-					} else
-						eprintf("Unknown syscall number\n");
-				} else {
-					n = rz_syscall_get_num(core->analysis->syscall, input + 2);
-					if (n != -1) {
-						rz_cons_printf("#define SYS_%s %s\n", input + 2, syscallNumber(n));
-					} else {
-						eprintf("Unknown syscall name\n");
-					}
-				}
-			} else {
-				list = rz_syscall_list(core->analysis->syscall);
-				rz_list_foreach (list, iter, si) {
-					rz_cons_printf("#define SYS_%s %s\n",
-						si->name, syscallNumber(si->num));
-				}
-				rz_list_free(list);
+RZ_IPI RzCmdStatus rz_analysis_syscall_name_handler(RzCore *core, int argc, const char **argv) {
+	st64 num = rz_syscall_get_num(core->analysis->syscall, argv[1]);
+	if (num < 1) {
+		RZ_LOG_ERROR("Cannot resolve syscall: %s\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println(syscallNumber(num));
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_syscall_number_handler(RzCore *core, int argc, const char **argv) {
+	st64 num = rz_num_math(core->num, argv[1]);
+	if (num < 1) {
+		RZ_LOG_ERROR("Cannot resolve syscall: %s\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzSyscallItem *si = rz_syscall_get(core->analysis->syscall, num, -1);
+	if (!si) {
+		RZ_LOG_ERROR("Cannot resolve syscall: %" PFMT64d "\n", num);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println(si->name);
+	return RZ_CMD_STATUS_OK;
+}
+
+static void syscall_dump(RzSyscallItem *si, bool is_c) {
+	if (is_c) {
+		rz_cons_printf("#define SYS_%s %s\n", si->name, syscallNumber(si->num));
+	} else {
+		rz_cons_printf(".equ SYS_%s %s\n", si->name, syscallNumber(si->num));
+	}
+}
+
+static RzCmdStatus syscalls_dump(RzCore *core, int argc, const char **argv, bool is_c) {
+	if (argc > 1) {
+		st64 n = rz_num_math(core->num, argv[1]);
+		if (n < 1) {
+			n = rz_syscall_get_num(core->analysis->syscall, argv[1]);
+			if (n == -1) {
+				RZ_LOG_ERROR("Cannot resolve syscall: %s\n", argv[1]);
+				return RZ_CMD_STATUS_ERROR;
 			}
+			rz_cons_printf(".equ SYS_%s %s\n", argv[1], syscallNumber(n));
```

Comment:
use `syscall_dump`

Reply:
I can't since it doesn't search the `RzSyscallItem *si` here, just the `char *` string.

Reply:
actually... can't we just remove these two lines of code and let the code fall through the `rz_syscall_get`, etc. ?

Reply:
@ret2libc at practice the logic between `rz_syscall_get_num()` and `rz_syscall_get()` is a bit different. I would prefer not to touch too much of this code until we decide to refactor the `librz/syscall/*` code which is a mess.